core/creators/themedui.js

```diff
@@ -384,6 +388,62 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayIfDetached` to be `true`.
+	//
+	// If `config.delayIfDetached_callback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayIfDetached: true,
+	//		delayIfDetached_interval: 50,
+	//		delayIfDetached_callback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {String} mode CKEditor mode.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	function delayCreationOnDetachedElement( element, mode, config, data ) {
+		if ( !config || !config.delayIfDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		if ( !config.delayIfDetached_callback ) {
+			CKEDITOR.config.delayIfDetached_callback( intervalyAttemptInstanceCreation );
+			return true;
+		}
+
+		config.delayIfDetached_callback( function() {
+			createInstance( element, config, data, mode );
+		} );
+
+		return true;
+
+		function intervalyAttemptInstanceCreation() {
+			var delay = parseFloat( config.delay );
+			delay = isNaN( delay ) ? CKEDITOR.config.delayIfDetached_interval : delay;
```

Comment:
This is a similar case we had with ensuring that element is detached after calling callback - if the config option is passed in the wrong format (e.g. `hello` instead of a number), we hide an error and not inform the developer about it making it very difficult to debug. We could throw a warning, but we don't do this for any other config option so I wouldn't introduce it now. Instead we should allow the script to fail and log the default browser debug tracing. So the default option should only be used if user doesn't provide any value.

Reply:
Also as for naming - delay is one-time thing and here we will use it for repeated action, so let's stick to the name `interval`.